src/rebar_core.erl

```diff
@@ -84,7 +84,7 @@ process_command(State, Command) ->
                     %% to have both $REBAR_PROFILE set and use 'as' in a command
                     case rebar_state:current_profiles(State) of
                         [default] ->
-                            do([{default, hd(TargetProviders)} | tl(TargetProviders)], State);
+                            do(TargetProviders, State);
```

Comment:
I think this is still needed. See without it:

```
→ DEBUG=1 ./rebar3 as test do help
===> Consult config file "rebar.config"
===> Consult config file "rebar.lock"
===> Uncaught error in rebar_core. Run with DEBUG=1 to see stacktrace
===> Uncaught error: {'EXIT',
                                {{badrecord,provider},
                                 [{providers,do,2,
                                      [{file,"src/providers.erl"},{line,86}]},
                                  {rebar_core,do,2,
                                      [{file,"src/rebar_core.erl"},
                                       {line,129}]},
                                  {rebar3,main,1,
                                      [{file,"src/rebar3.erl"},{line,46}]},
                                  {escript,run,2,
                                      [{file,"escript.erl"},{line,752}]},
                                  {escript,start,1,
                                      [{file,"escript.erl"},{line,276}]},
                                  {init,start_it,1,[]},
                                  {init,start_em,1,[]}]}}
```

Probably needs a test case for that too